dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/tasks/Ds3PutJob.java

```diff
@@ -32,297 +29,210 @@
+                ds3Client.getConnectionDetails().getEndpoint(), this.getJobId(), totalJobSize, targetDir, PUT, bucket);
+
+        updateMessage(StringBuilderUtil.transferringTotalJobString(FileSizeFormat.getFileSizeType(totalJobSize), targetDirectory).toString());
+
+        if (hasPriority) {
+            ds3Client.modifyJobSpectraS3(new ModifyJobSpectraS3Request(job.getJobId()).withPriority(Priority.valueOf(jobPriority)));
+        }
+
+        if (metadata) {
+            LOG.info("Registering metadata access Implementation");
+            job.withMetadata(new MetadataAccessImpl(fileMapper));
         }
+
+        addWaitingForChunkListener(totalJobSize, bucket + StringConstants.DOUBLE_SLASH + targetDir);
+        final AtomicLong totalSent = addDataTransferListener(totalJobSize);
+        job.attachObjectCompletedListener(obj -> updateGuiOnComplete(startJobDate, jobStartInstant, totalJobSize, totalSent, obj));
+
+        job.transfer(file -> FileChannel.open(PathUtil.resolveForSymbolic(fileMapper.get(file)), StandardOpenOption.READ));
+        waitForPermanentStorageTransfer(totalJobSize);
+        ParseJobInterruptionMap.removeJobID(jobInterruptionStore, this.getJobId().toString(), ds3Client.getConnectionDetails().getEndpoint(), deepStorageBrowserPresenter, loggingService);
     }
 
-    /**
-     * Create map of folders from the selected list
-     *
-     * @param directories      selected directories
-     * @param expandedPaths    expandedPaths
-     * @param partOfDirBuilder partOfDirBuilder
-     */
-    public ImmutableMap.Builder<String, Path> createFolderMap(final ImmutableList<Path> directories,
-                                                              final ImmutableMultimap.Builder<Path, Path> expandedPaths,
-                                                              final ImmutableSet.Builder<Path> partOfDirBuilder) {
-        final ImmutableMap.Builder<String, Path> folderMap = ImmutableMap.builder();
-        directories.forEach(path -> {
+    private static void buildMaps(final ImmutableMap.Builder<String, Path> fileMapBuilder, final ImmutableMap.Builder<String, Path> folderMapBuilder, final Pair<String, Path> pair, final LoggingService loggingService) {
+        final String name = pair.getKey();
+        final Path path = pair.getValue();
+        if (Files.isDirectory(pair.getValue())) {
             try {
-                partOfDirBuilder.add(path);
-                expandedPaths.putAll(path, Files.walk(path).filter(child -> !Files.isDirectory(child)).collect(GuavaCollectors.immutableList()));
-                final String ds3ObjPath = getDs3ObjectPath(path, path, true, files.size(), directories.size());
-                final String ds3FileName = PathUtil.toDs3Path(targetDir, ds3ObjPath);
-                folderMap.put(ds3FileName, path);
+                Files.walk(path).filter(child -> !Files.isDirectory(child)).map(p -> new Pair<>(name + "/" + path.relativize(p).toString(), p))
```

Comment:
Should we use the helper functions to do this or is there some custom logic in here that's different from the helper functions?

Reply:
I might be missing one, but as far as I can tell all of the helper functions give me DS3Objects, but for this we need the intermediate map.